module/blobs/blob_io.go

```diff
@@ -0,0 +1,313 @@
+package blobs
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+// This file contains the implementation of a Blob Channel,
+// which represents an in-memory pipe for blob data.
+//
+// It is safe for the sender and receiver objects returned
+// from IncomingBlobChannel or OutgoingBlobChannel to be used
+// in parallel with each other. However, it is not safe to use
+// the same sender / receiver in parallel.
```

Comment:
adding some more depth
```suggestion
// This file contains the implementation of a Blob Channel, which represents an in-memory pipe for
// blob data.
//
// It is safe for the sender and receiver objects returned from IncomingBlobChannel or
// OutgoingBlobChannel to be used in parallel with each other. However, it is not safe to use the
// same sender / receiver in parallel.
//
// Typically, interactions with these objects are done using IncomingBlobChannel and OutgoingBlobChannel.
//
// IncomingBlobChannel is used when adding new blobs, and returns a BlobChannelWriter and BlobReceiver.
// Data is serialized and written to the BlobChannelWriter, which breaks the data into blobs and passes
// them to the BlobReceiver. The BlobReceiver is used by the caller to get a stream of all blobs written. 
//
// OutgoingBlobChannel is used when reading existing blobs, and returns a BlobChannelReader and BlobSender.
// Blobs are read and sent through BlobSender to the BlobChannelReader. The BlobChannelReader aggregates 
// the blob data so it can be deserialized back into its original format.
```